### PR TITLE
Add VCR recording and rerecording documentation

### PR DESCRIPTION
**Pull request by jsugarman**
## What
Document VCR cassette recording and re-recording

Came out of me trying to rerecord all cassettes and the
difficulties for local env var setup in particular.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: docs/vcr_recording.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ - You will need to set the following env vars in `.env.test` to use the staging environments details.
+
+  ```
+  BC_LSC_SERVICE_NAME=
+  BC_CLIENT_ORG_ID=
+  BC_CLIENT_USER_ID=
+  BC_WSDL_URL=
+  BC_USE_DEV_MOCK=false
+  ```
+ - Important: turn off the benefit checker mock (`BC_USE_DEV_MOCK=false`) when recording, but switch back on later (does it matter?)
```
Comment (willc-work) [inline docs/vcr_recording.md:17]: might be good to say, 'switch BC_USE_DEV_MOCK to false, if it is enabled, and use the staging benefit checker'. e.g. I always use the mock service locally but you could just as easily use the staging benefit checker and never use the mock service

**Review by willc-work — APPROVED** [review]
just one comment about BenChecker

File: docs/vcr_recording.md (hunk 1 above)
Comment (jsugarman) [reply docs/vcr_recording.md]: have tweaked that now @willc-work

File: README.md [hunk 2]
```diff
@@ -223,9 +223,10 @@ Runs Rubocop, RSpec specs and Cucumber features
 bin/rake
 ```
 
-**NOTE:** To ensure the recorded VCRs used in the specs are up to date, run the tests as such:
+**NOTE:** To ensure the recorded VCRs used in the specs are up to date, run the tests as such below. However, see [VCR Recording](docs/vcr_recording.md) for setup and recommended approaches to rerecording of cassettes.
 
-```
+```sh
+# Not recommended
```
[the remark this answers is not in the document]
Comment (jsugarman) [reply README.md:229]: 👍 okey dokey

File: docs/vcr_recording.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  BC_CLIENT_USER_ID=
+  BC_WSDL_URL=
+  BC_USE_DEV_MOCK=false
+  ```
+
+ - ensure you switch `BC_USE_DEV_MOCK=false` when recording. You can switch `BC_USE_DEV_MOCK=true` after recording or leave to use staging.
+
+### Provider manager issues
+  - connect to VPN
+  - set PROVIDER_MANAGER_URL
```
Comment (willc-work) [inline docs/vcr_recording.md:22]: should this be PROVIDER_DETAILS_URL

Comment (jsugarman) [reply docs/vcr_recording.md]: corrected

**Pull request merged**